src/main/java/seedu/address/MainApp.java

```diff
@@ -83,11 +84,17 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
                 logger.info("Creating a new data file " + storage.getAddressBookFilePath()
                         + " populated with a sample AddressBook.");
             }
-            initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
+            // line below can throw SampleDataLoadingException
+            ReadOnlyAddressBook sampleAddressBook = SampleDataUtil.getSampleAddressBook();
+            initialData = addressBookOptional.orElse(sampleAddressBook);
```

Comment:
Just to confirm this is just a refactoring of getting the AddressBook based off the SampleDataUtil right?

Reply:
This is because java will throw an error if I set the function that throws the exception in the orElseGet method.